### PR TITLE
Update TEXT with code changes

### PR DESCRIPTION
**Pull request by mschmo**
This updates TEXT.md with the code changes from PRs #22, #23 and #24

### CODE REVIEW

File: TEXT.md [hunk 1]
```diff
@@ -164,9 +164,9 @@ let message = realstring.push_str(str1);
 - add `&str`s using `format!`
 
 ```rust
-let str1 = "hello ";
-let str2 = "world!";
-let message = format!("{}{}", str1, str2);
+let mut realstring = String::from("hello ");
```
Comment (ashleygwilliams) [inline TEXT.md]: this example is supposed to use format and doesn't. i can't accept this change.

File: TEXT.md [hunk 2]
```diff
@@ -181,7 +181,8 @@ let message = format!("{}{}", str1, str2);
 Example:
 
 ```rust
-let letters = String::from("ashley").chars();
+let name = String::from("ashley");
+let letters = name.chars();
```
Comment (ashleygwilliams) [inline TEXT.md:185]: 👍

File: TEXT.md [hunk 3]
```diff
@@ -218,7 +219,7 @@ format!("my dogs are named: {} and {}", "cheeto", "frito");
 ## function signatures
 
 ```rust
-pub fn say_hello(name: &str) -> &str {
+pub fn say_hello(name: &str) -> String {
```
Comment (ashleygwilliams) [inline TEXT.md:222]: 👍

File: TEXT.md [hunk 4]
```diff
@@ -264,7 +265,7 @@ a Type inside, e.g. `Option<u32>`.
 For example, if a parameter is optional you'd write:
 
 ```rust
-fn greeting(name: Option<&str>) -> &str {
+fn greeting(name: Option<&str>) -> String {
```
Comment (ashleygwilliams) [inline TEXT.md:268]: 👍

File: TEXT.md [hunk 5]
```diff
@@ -306,7 +307,7 @@ inline with your code.
 To designate a test write `#[test]` above a code block with asserts:
 
 ```rust
-fn say_hello(name: &str) -> &str {
+fn say_hello(name: &str) -> String {
```
Comment (ashleygwilliams) [inline TEXT.md:310]: 👍

File: TEXT.md [hunk 6]
```diff
@@ -316,9 +317,9 @@ fn say_hello(name: &str) -> &str {
 
 #[test]
 fn it_should_say_hello() {
-  assert_eq!(say_hello(None), "Hello, World!");
-  assert_ne!(say_hello(Some("ashley")), "Hello, World!");
-  assert_eq!(say_hello(Some("ashley")), "Hello, ashley!");
+  assert_eq!(say_hello(None), String::from("Hello, World!"));
+  assert_eq!(say_hello(Some("World")), String::from("Hello, World!"));
+  assert_eq!(say_hello(Some("ashley")), String::from(("Hello, ashley!"));
```
Comment (ashleygwilliams) [inline TEXT.md:322]: 👍

**Review by ashleygwilliams — CHANGES_REQUESTED** [review]
will accept if you remove the first change. thanks!

File: TEXT.md (hunk 1 above)
Comment (ashleygwilliams) [reply TEXT.md]: i think you meant for these changes to be in the above example (based on your other PRs!)

Comment (mschmo) [reply TEXT.md]: Ah yeah my bad. That's why I shouldn't submit PRs while hungover. I'll fix it.

**Review by ashleygwilliams — APPROVED** [review]

**Pull request merged**